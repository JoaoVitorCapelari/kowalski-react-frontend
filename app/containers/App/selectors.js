/**
 * The global state selectors
 */

import { createSelector } from 'reselect';

const selectGlobal = (state) => state.get('global');

const selectRoute = (state) => state.get('route');

const makeSelectCurrentUser = () => createSelector(
  selectGlobal,
  (globalState) => globalState.get('currentUser')
);

const makeSelectLocation = () => createSelector(
  selectRoute,
  (routeState) => routeState.get('location').toJS()
);

const makeSelectPageBeforeAuthError = () => createSelector(
  selectGlobal,
  (globalState) => globalState.get('pageBeforeAuthError'),
);

const makeSelectActivePage = () => createSelector(
  makeSelectLocation(),
  (location) => {
    const path = location.pathname;
    if (path.match(/projects/gi)) {
      return 'projects';
    } else if (path.match(/people/gi)) {
      return 'people';
    } else if (path.match(/timesheet/gi)) {
      return 'timesheet';
    }
    throw new Error(`Unmachable path: ${path}`);
  }
);

export {
  selectGlobal,
  selectRoute,
  makeSelectCurrentUser,
  makeSelectLocation,
  makeSelectPageBeforeAuthError,
  makeSelectActivePage,
};
