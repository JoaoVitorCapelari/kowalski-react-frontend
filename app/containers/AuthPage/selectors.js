import { createSelector } from 'reselect';
import { fromJS } from 'immutable';

/**
 * Direct selector to the authPage state domain
 */
const selectAuthPageDomain = (state) => state.get('authpage') || fromJS({});

/**
 * Other specific selectors
 */


/**
 * Default selector used by AuthPage
 */

const makeSelectAuthPage = () => createSelector(
  selectAuthPageDomain,
  (substate) => substate.toJS()
);

const makeSelectUsername = () => createSelector(
  selectAuthPageDomain,
  (substate) => substate.get('username'),
);

const makeSelectPassword = () => createSelector(
  selectAuthPageDomain,
  (substate) => substate.get('password'),
);

const makeSelectIsSubmitting = () => createSelector(
  selectAuthPageDomain,
  (substate) => substate.get('isSubmitting'),
);

const makeSelectServerErrorMsg = () => createSelector(
  selectAuthPageDomain,
  (substate) => substate.get('serverErrorMsg'),
);

export default makeSelectAuthPage;
export {
  selectAuthPageDomain,
  makeSelectUsername,
  makeSelectPassword,
  makeSelectIsSubmitting,
  makeSelectServerErrorMsg,
};
