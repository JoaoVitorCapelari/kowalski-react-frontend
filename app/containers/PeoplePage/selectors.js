import { createSelector } from 'reselect';
import { fromJS, Map } from 'immutable';

/**
 * Direct selector to the peoplePage state domain
 */
const selectPeoplePageDomain = (state) => state.get('peoplePage') || Map();

/**
 * Other specific selectors
 */


/**
 * Default selector used by PeoplePage
 */

const makeSelectPeoplePage = () => createSelector(
  selectPeoplePageDomain,
  (substate) => substate.toJS()
);

const makeSelectPeople = () => createSelector(
  selectPeoplePageDomain,
  (substate) => substate.get('people') || fromJS([])
);

const makeSelectIsNewPersonFormOpen = () => createSelector(
  selectPeoplePageDomain,
  (substate) => substate.get('personFormOpen'),
);

const makeSelectIsSubmittingNewPerson = () => createSelector(
  selectPeoplePageDomain,
  (substate) => substate.get('isSubmittingNewPersonForm'),
);

const makeSelectIsLoadingPeople = () => createSelector(
  selectPeoplePageDomain,
  (substate) => substate.get('isLoadingPeople'),
);

const makeSelectLoadingPeopleErrorMsg = () => createSelector(
  selectPeoplePageDomain,
  (substate) => substate.get('loadingPeopleErrorMsg'),
);

const makeSelectNewPersonFormSubmissionError = () => createSelector(
  selectPeoplePageDomain,
  (substate) => substate.get('newPersonFormSubmissionError'),
);

export {
  selectPeoplePageDomain,
  makeSelectPeople,
  makeSelectIsNewPersonFormOpen,
  makeSelectIsSubmittingNewPerson,
  makeSelectIsLoadingPeople,
  makeSelectLoadingPeopleErrorMsg,
  makeSelectNewPersonFormSubmissionError,
};

export default makeSelectPeoplePage;
