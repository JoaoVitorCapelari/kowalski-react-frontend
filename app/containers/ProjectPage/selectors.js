import { createSelector } from 'reselect';
import { fromJS, List } from 'immutable';

/**

 * Direct selector to the projectPage state domain

 */

const selectProjectPageDomain = (state) => state.get('projectpage') || fromJS({});

/**

 * Other specific selectors

 */

const makeSelectProjectCodes = () => createSelector(
  selectProjectPageDomain,
  (substate) => substate.get('projectCodes')
);


/**

 * Default selector used by ProjectPage

 */

const makeSelectProjectPage = () => createSelector(
  selectProjectPageDomain,
  (substate) => substate.toJS()
);

const makeSelectSelectedProjectCode = () => createSelector(
  selectProjectPageDomain,
  (substate) => substate.get('selectedProjectCode'),
);

const makeSelectSelectedProject = () => createSelector(
  selectProjectPageDomain,
  (substate) => substate.get('selectedProject'),
);

const makeSelectSelectedTab = () => createSelector(
  selectProjectPageDomain,
  (substate) => substate.get('selectedTab'),
);

const makeSelectActivityFilteringText = () => createSelector(
  selectProjectPageDomain,
  (substate) => substate.get('activityFilteringText'),
);

const makeSelectExpandedTaskIds = () => createSelector(
  selectProjectPageDomain,
  (substate) => substate.get('expandedActivityIds'),
);

const makeSelectIsNewTaskFormDialogOpened = () => createSelector(
  selectProjectPageDomain,
  (substate) => substate.get('isNewTaskFormDialogOpened'),
);

const makeSelectActivityLoadedIntoNewTaskForm = () => createSelector(
  selectProjectPageDomain,
  (substate) => substate.get('activityLoadedIntoNewTaskForm'),
);

const makeSelectIsNewActivityFormDialogOpened = () => createSelector(
  selectProjectPageDomain,
  (substate) => substate.get('isNewActivityFormDialogOpened'),
);

const makeSelectProjectLoadedIntoNewTaskForm = () => createSelector(
  selectProjectPageDomain,
  (substate) => substate.get('projectLoadedIntoNewTaskForm'),
);

const makeSelectProjectLoadedIntoNewActivityForm = () => createSelector(
  selectProjectPageDomain,
  (substate) => substate.get('projectLoadedIntoNewActivityForm'),
);

const makeSelectIsSubmittingNewActivity = () => createSelector(
  selectProjectPageDomain,
  (substate) => substate.get('isSubmittingNewActivity'),
);

const makeSelectIsSubmittingNewTask = () => createSelector(
  selectProjectPageDomain,
  (substate) => substate.get('isSubmittingNewTask'),
);

const makeSelectLoadingProjectCodesError = () => createSelector(
  selectProjectPageDomain,
  (substate) => substate.get('loadingProjectCodesError'),
);

const makeSelectLoadingProjectError = () => createSelector(
  selectProjectPageDomain,
  (substate) => substate.get('loadingProjectError'),
);

const makeSelectIsAddPeopleFormOpen = () => createSelector(
  selectProjectPageDomain,
  (substate) => substate.get('isAddPeopleFormOpen'),
);

const makeSelectUsers = () => createSelector(
  selectProjectPageDomain,
  (substate) => substate.get('users'),
);

const makeSelectUserIdsInProject = () => createSelector(
  makeSelectSelectedProject(),
  (selectedProject) => (
    selectedProject
       ? selectedProject.get('people').map((person) => person.get('kUserId'))
       : List()
  ),
);

const makeSelectLoadingUsersError = () => createSelector(
  selectProjectPageDomain,
  (substate) => substate.get('loadingUsersError'),
);

const makeSelectUsersInProject = () => createSelector(
  [makeSelectUserIdsInProject(), makeSelectUsers()],
  (userIdsInProject, users) => users.filter((user) => userIdsInProject.includes(user.get('kUserId'))),
);

const makeSelectUsersNotInProject = () => createSelector(
  [makeSelectUserIdsInProject(), makeSelectUsers()],
  (userIdsInProject, users) => users.filter((user) => !userIdsInProject.includes(user.get('kUserId'))),
);

export default makeSelectProjectPage;

export {
  selectProjectPageDomain,
  makeSelectProjectCodes,
  makeSelectSelectedProjectCode,
  makeSelectSelectedProject,
  makeSelectSelectedTab,
  makeSelectActivityFilteringText,
  makeSelectExpandedTaskIds,
  makeSelectIsNewTaskFormDialogOpened,
  makeSelectActivityLoadedIntoNewTaskForm,
  makeSelectIsNewActivityFormDialogOpened,
  makeSelectProjectLoadedIntoNewActivityForm,
  makeSelectIsSubmittingNewActivity,
  makeSelectIsSubmittingNewTask,
  makeSelectProjectLoadedIntoNewTaskForm,
  makeSelectLoadingProjectCodesError,
  makeSelectLoadingProjectError,
  makeSelectIsAddPeopleFormOpen,
  makeSelectUsersInProject,
  makeSelectUsersNotInProject,
  makeSelectLoadingUsersError,
};

