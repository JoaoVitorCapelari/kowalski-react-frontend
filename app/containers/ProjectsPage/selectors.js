import { createSelector } from 'reselect';
import { fromJS } from 'immutable';

/**
 * Direct selector to the projectsPage state domain
 */
const selectProjectsPageDomain = (state) => state.get('projectspage') || fromJS({});

/**
 * Other specific selectors
 */


/**
 * Default selector used by ProjectsPage
 */

const makeSelectProjectsPage = () => createSelector(
  selectProjectsPageDomain,
  (substate) => substate.toJS()
);

const makeSelectProjects = () => createSelector(
  selectProjectsPageDomain,
  (substate) => substate.get('projects')
);

const makeSelectIsNewProjectFormOpen = () => createSelector(
  selectProjectsPageDomain,
  (substate) => substate.get('projectFormOpen'),
);

const makeSelectIsSubmittingNewProject = () => createSelector(
  selectProjectsPageDomain,
  (substate) => substate.get('isSubmittingNewProjectForm'),
);

export default makeSelectProjectsPage;
export {
  selectProjectsPageDomain,
  makeSelectProjects,
  makeSelectIsNewProjectFormOpen,
  makeSelectIsSubmittingNewProject,
};
