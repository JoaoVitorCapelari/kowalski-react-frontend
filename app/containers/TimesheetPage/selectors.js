import { createSelector } from 'reselect';
import { Map, List } from 'immutable';
import startOfWeek from 'date-fns/start_of_week';
import endOfWeek from 'date-fns/end_of_week';
import format from 'date-fns/format';
import { DATE_DAY_FORMAT } from './constants';

/**
 * Direct selector to the timesheetPage state domain
 */
const selectTimesheetPageDomain = (state) => state.get('timesheetpage') || Map();

const makeSelectSelectedDate = createSelector(
  [selectTimesheetPageDomain],
  (substate) => substate.get('selectedDate'),
);

const makeSelectSelectedRange = createSelector(
  [makeSelectSelectedDate],
  (selectedDate) => {
    const extractedStartOfWeek = startOfWeek(selectedDate);
    const extractedEndOfWeek = endOfWeek(selectedDate);
    return [extractedStartOfWeek, extractedEndOfWeek];
  }
);

const makeSelectIsSubmitting = createSelector(
  [selectTimesheetPageDomain],
  (substate) => substate.get('isSubmitting'),
);

const makeTimeSlotEntriesSelector = createSelector(
  [selectTimesheetPageDomain],
  (substate) => substate.get('timeSlotEntries') || List(),
);

const makeTimeSlotDayMapSelector = createSelector(
  makeTimeSlotEntriesSelector,
  (timeSlotEntries) => {
    const result = timeSlotEntries.reduce((accumulator, timeSlotEntry) => {
      const dateDayFormat = format(timeSlotEntry.get('day'), DATE_DAY_FORMAT);
      return accumulator.set(
        dateDayFormat,
        (accumulator.get(dateDayFormat) || List()).push(timeSlotEntry)
      );
    }, Map());
    return result;
  },
);

export {
  selectTimesheetPageDomain,
  makeSelectSelectedDate,
  makeSelectSelectedRange,
  makeSelectIsSubmitting,
  makeTimeSlotEntriesSelector,
  makeTimeSlotDayMapSelector,
};
