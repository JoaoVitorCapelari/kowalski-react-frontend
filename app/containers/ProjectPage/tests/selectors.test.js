import { fromJS } from 'immutable';
import { selectProjectPageDomain } from '../selectors';

describe('selectProjectPageDomain', () => {
  it('selects the proper substate', () => {
    const state = fromJS({
      testing: 'a',
      projectspage: {
        testing2: 'b',
      },
    });
    expect(selectProjectPageDomain(state)).toMatchSnapshot();
  });
});
