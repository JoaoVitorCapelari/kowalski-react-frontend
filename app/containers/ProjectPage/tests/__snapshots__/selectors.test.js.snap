// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`selectProjectPageDomain selects the proper substate 1`] = `
Immutable.Map {
}
`;
